### ISSUE TITLE
Issues Observed During Unimodal Pre-training and Multimodal Fine-tuning

### ISSUE THREAD

**Issue by sdeerman** [gw-zhong/CMC#4]
Dear gw-zhong,

Hello, and thank you again for your previous reply. I have now been able to reproduce most of the results reported in your paper, with only minor discrepancies that remain within an acceptable range. However, during the reproduction process I encountered a few additional questions, and I would like to take a bit of your valuable time to seek your advice.

1) About the unimodal pre-training stage:
During unimodal training, the accuracies of the three modalities tend to stay around a fixed value and appear almost as horizontal lines in the training curves. The validation and test losses also tend to converge after around 60 epochs. I would like to ask whether this behavior is consistent with what you observed in your experiments. In your paper, aside from Table VI, I did not find other detailed training results related to the unimodal stage.

2) About the multimodal fine-tuning stage:
In the fine-tuning command provided in your README, the learning rate parameter is not explicitly specified, so the model uses the default value of 1e-3 during training. After I changed the learning rate to 5e-6, I was able to largely reproduce the results reported in your paper. However, I noticed a phenomenon similar to point (1): metrics such as Has0_acc_2 and Has0_F1_score remain nearly constant during training, forming horizontal lines on the curve. In some cases, the metrics even degrade as training continues.

I would like to ask whether this behavior might be related to the small remaining discrepancies between my reproduced results and those in your paper. If the purpose of multimodal fine-tuning is mainly to optimize the MCR module parameters, why is there no obvious improvement in performance during the fine-tuning stage?

Thank you very much for your work and for taking the time to answer my questions. I look forward to your reply.

Best regards

PS. In the attachment, I have uploaded the training curves corresponding to the issues I described. I hope they may be helpful for your analysis.

<img width="1053" height="744" alt="Image" src="https://github.com/user-attachments/assets/3a608ae2-4835-4f64-8888-e91f469ac17e" />
<img width="1053" height="747" alt="Image" src="https://github.com/user-attachments/assets/1cd4cd96-fbfc-423b-b629-e9cae9f1ffc5" />

**Comment by gw-zhong** [comment]
Hi, we are delighted to hear that you are able to reproduce our work! Here are our detailed responses to your questions:

1) Yes, your observation is very keen. For small datasets like SIMS, unimodal models often converge within just a few epochs. If you calculate the accuracy using the true values of unimodal labels during training, you might see that their accuracy decreases as training progresses. This is because they are very prone to overfitting to pseudo-labels. However, pseudo-labels themselves contain noise (since pseudo-labels are not 100% equal to the true unimodal labels). Therefore, overfitting to pseudo-labels is actually a harmful behavior. In our paper, we used the EMA method to smooth and save the model, so that the model itself does not overfit to the pseudo-labels of the validation set as much as possible.

2) Similarly, during the multimodal fine-tuning stage, for smaller datasets, since the weights of the unimodal models are pre-trained, the model can converge within just a few epochs. In subsequent training, the model may no longer focus on optimizing the task loss, but rather on optimizing the contrastive loss (i.e., the supcon loss), leading to fluctuations in accuracy.

3) Regarding the MCR issue, you can refer to Table V in our paper, where we have conducted a detailed discussion. Our conclusion is that when the features of non-text modalities have a certain degree of reliability (for example, the visual modality in the SIMS and SIMS-v2 datasets), training only MCR during the multimodal fine-tuning stage while freezing all other parameters (i.e., the `CMC-variant` row) can achieve higher performance improvements. However, when the features of non-text modalities are not that reliable (for example, in MOSI and MOSEI), training only MCR may actually lead to a decline in performance (because it is equivalent to lowering the reliability of the most reliable text modality and increasing the reliability of the unreliable non-text modalities). Therefore, to make the entire model more robust, we decided to further train all parameters of the model during the multimodal fine-tuning stage to adapt to various datasets. Additionally, the `w/o MCR` row also shows that when training with multimodal labels for supervision, the model's performance will decline without MCR adjusting at the decision level.

Finally, we suggest that you can try training only MCR on the SIMS and SIMS-v2 datasets, freezing all other parameters and using only multimodal labels for supervision (after freezing the unimodal models, the unimodal pseudo-labels will not be involved in the loss calculation, as there is no gradient). You may achieve further performance improvements.

Best wishes
